internal/dag/policy.go

```diff
@@ -0,0 +1,39 @@
+// Copyright © 2019 Heptio
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dag
+
+import (
+	"time"
+
+	"github.com/heptio/contour/apis/contour/v1beta1"
+)
+
+func retryPolicyIngressRoute(rp *v1beta1.RetryPolicy) *RetryPolicy {
+	if rp == nil {
+		return nil
+	}
+	perTryTimeout, _ := time.ParseDuration(rp.PerTryTimeout)
```

Comment:
Shouldn't you check the error returned from `time.ParseDuration` call?

Reply:
It's not specified in the documentation that ParseDuration returns a zero duration in an error condition, but its been like that since pre Go 1.0 so we can utilise this here. We have tests that will catch if the behaviour of ParseDuration ever changes -- not that it will.